activestorage/lib/active_storage/service/configurator.rb

```diff
@@ -9,7 +9,7 @@ def self.build(service_name, configurations)
     end
 
     def initialize(configurations)
-      @configurations = configurations.deep_symbolize_keys
+      @configurations = build_configurations(configurations)
```

Comment:
Think better to split `build_configurations` into something like `configuration_from_url(configurations) || configurations.deep_symbolize_keys`. Then you make the `configuration_from_url` do only one thing and you relieve yourself of having to do exit-wound control flow.